README.md

```diff
@@ -57,48 +60,253 @@ This is work in progress as of 14 Nov 2018:
 [![License](https://img.shields.io/npm/l/@airtasker/spot.svg)](https://github.com/airtasker/spot/blob/master/package.json)
 
 <!-- toc -->
-* [Usage](#usage)
-* [Commands](#commands)
-<!-- tocstop -->
+
+- [Usage](#usage)
+- [Commands](#commands)
+  <!-- tocstop -->
+
 # Usage
 
 To get started and set up an API declaration in the current directory, run:
+
 ```
 npx @airtasker/spot init
 ```
 
 You can then run a generator with:
+
 ```
 npx @airtasker/spot generate --api api.ts
 ```
 
+## `@api`
+
+Define an API. This is required and must only be defined once:
+
+```TypeScript
+import { api } from "@airtasker/spot";
+
+@api({
+  name: "My API",
+  description: "My really cool API"
+})
+class MyAPI {}
+```
+
+| Field         | Description                           |
+| ------------- | ------------------------------------- |
+| `name`        | (**required**) Name of the API        |
+| `description` | (**required**) Description of the API |
+
+## `@endpoint`
+
+Define a HTTP endpoint for the API. An endpoint describes a particular HTTP action on a URL path:
+
+```TypeScript
+import { endpoint, genericError, header, pathParam, queryParam, request, response, specificError } from "@airtasker/spot";
+
+class MyUserEndpoints {
+  @endpoint({
```

Comment:
It may be useful to add a comment before each of these endpoint definitions, paraphrasing what the declaration does.

```suggestion
  // GET /users expects a mandatory `search_term` query parameter and returns a list of users.
  @endpoint({
```

Note #1: I wrote "mandatory", but I wonder if `search_term` should be `Optional<string>` to reflect real endpoints more accurately? It's rare for a search query parameter to be mandatory.

Note #2: Should we instead return a `UserListResponse`? Returning a list of responses doesn't feel very idiomatic.